addons/discuss-button/userscript.js

```diff
@@ -1,29 +1,50 @@
 export default async function ({ addon, global, console }) {
-  if (!addon.tab.clientVersion) return;
-  if (document.location.pathname.startsWith("/conference")) return;
+  const searchItem = await addon.tab.waitForElement(".search", {
+    markAsSeen: true,
+    reduxCondition: (state) => {
+      if (!state.scratchGui) return true;
+      return state.scratchGui.mode.isPlayerOnly;
+    },
+  });
+  const scratchr2List = document.querySelector(".site-nav");
 
-  const link = document.createElement("li");
-  link.className = "link discuss";
-  const a = document.createElement("a");
-  a.href = "/discuss";
-  a.textContent = addon.settings.get("buttonName");
-  link.appendChild(a);
+  function createItem({ name, url }, last) {
+    const li = document.createElement("li");
+    li.className = "link";
+    if (last) li.classList.add("last");
 
-  if (addon.tab.clientVersion === "scratch-www") {
-    while (true) {
-      const el = await addon.tab.waitForElement("div#navigation div.inner ul:not(.dropdown)", {
-        markAsSeen: true,
-        reduxCondition: (state) => {
-          if (!state.scratchGui) return true;
-          return state.scratchGui.mode.isPlayerOnly;
-        },
-      });
-      if (addon.settings.get("removeIdeasBtn")) el.getElementsByTagName("li")[3].remove();
-      el.insertBefore(link, el.getElementsByTagName("li")[3]);
+    const a = document.createElement("a");
+    a.href = url.startsWith("javascript") ? "" : url;
```

Comment:
```suggestion
    a.href = /^https?:\/\//g.test(url) ? url : "";
```

Reply:
What if I wanted to put the sa settings page url? Why not just exclude js urls?

Reply:
```suggestion
    a.href = /^(?:https?|(?:chrome|moz)-extension):\/\//g.test(url) ? url : "";
```

Reply:
> What if I wanted to put the sa settings page url? Why not just exclude js urls?

I don't think scratch.mit.edu can directly link to chrome-extension://*

Reply:
@apple502j Your commit has broken pathnames.

Reply:
Does this work ok?
```suggestion
    a.href = /\s*javascript:/g.test(url) ? "" : url;
```

Reply:
FYI Apple, on edge it's just `extention://` without the prefix

But why exclude js urls anyway?

Reply:
Yeah keeping it would be cool. Tho it's a bad idea for many reasons.